oak_restricted_kernel/src/dice_attestation.rs

```diff
@@ -15,37 +15,24 @@
 //
 
 use coset::CborSerializable;
-use zerocopy::FromZeroes;
+use oak_crypto::hpke::Serializable;
 
-/// Generates an ECA certificate for use by the application.
-fn generate_application_certificate(
-    kernel_signing_key: &p256::ecdsa::SigningKey,
-    kernel_cert_issuer: alloc::string::String,
-    application_verifying_key: &p256::ecdsa::VerifyingKey,
-    app_digest: &[u8],
-) -> coset::CoseSign1 {
-    let additional_claims = alloc::vec![(
-        coset::cwt::ClaimName::PrivateUse(oak_dice::cert::LAYER_2_CODE_MEASUREMENT_ID),
-        coset::cbor::value::Value::Bytes(app_digest.into()),
-    )];
-    oak_dice::cert::generate_signing_certificate(
-        kernel_signing_key,
-        kernel_cert_issuer,
-        application_verifying_key,
-        additional_claims,
-    )
-    .expect("couldn't generate signing certificate")
+fn certificate_to_byte_array(cert: coset::CoseSign1) -> [u8; oak_dice::evidence::CERTIFICATE_SIZE] {
+    let vec = cert.to_vec().expect("couldn't serialize certificate");
+    let mut slice = [0; oak_dice::evidence::CERTIFICATE_SIZE];
```

Comment:
You should be able to turn a vec into an array with just `try_into()` (although it fails if the sizes don't match -- but this code would panic if vec is any longer than CERTIFICATE_SIZE anyway)

Reply:
it wouldn't panic, since we copy_from_slice on a subslice of of the length `vec.len()`. 

I we could use try_into to avoid the surplus copy in case they do do happen to match in length. 

@conradgrobler do we expect the CoseSign1 cert to have consistent length?

Reply:
`slice` has size of `CERTIFICATE_SIZE`; if `vec.len()` > `CERTIFICATE_SIZE` then `slice[..vec.len()]` will panic.

Reply:
The existing code has the benefit of not failing if `vec.len()` < `CERTIFICATE_SIZE` (with zero padding); if we want truncation, then the slicing op should be `slice[..vec.len().min(CERTIFICATE_SIZE)]`. But I don't know whether we want that (probably not); in any case, if we're fine with a panic, then I'd add an explicit check and `panic!()` with a more descriptive error message than just "slice index out of bounds" or whatever the rather nondescript message is that you'd get.

Reply:
fair. We should do the same in stage0 then, which uses the same logic as this PR. Shared functions to handle encoding and decoding would be a good idea, and would handle this in a consistent manner. I'll create a follow up task.

Reply:
> @conradgrobler do we expect the CoseSign1 cert to have consistent length?

It don't think a fixed length is guaranteed, but I assumed that it would always be shorter than the array size. If it is longer than the array size we should either return an error or panic since the certificate will be truncated and therefore attestation verification will fail, so it it not useful to continue booting. Additional context on in such an error case would be useful (in a future PR).